### PR TITLE
Add UUIDs to SDFG elements

### PR DESCRIPTION
**Pull request by phschaad**

### CODE REVIEW

**Comment by tbennun** [comment]
The diff looks good, the error is weird (because `super().__init__()` is called). I would refactor the uuid function to a central location so that it is configurable (e.g., for determinism).

**Comment by phschaad** [comment]
I tried debugging the error a bit and I am genuinely super confused what is going on here. It doesn't even matter what I name the property, it really is like it is never defined for some instances of objects, but it is for others - despite `super().__init__()` seemingly being called everywhere, as you state.

**Comment by phschaad** [comment]
It's `__deepcopy__`...

**Comment by tbennun** [comment]
@phschaad so we already have a validation check that tests the uniqueness of `id(obj)` for objects for bad references. I wonder if we should also verify that for `obj.id` (probably secondary)

File: dace/sdfg/nodes.py [hunk 1]
```diff
@@ -574,6 +580,9 @@ def __deepcopy__(self, memo):
         result = cls.__new__(cls)
         memo[id(self)] = result
         for k, v in self.__dict__.items():
+            # Skip GUID.
+            if k in ('guid'):
```
Comment (tbennun) [inline dace/sdfg/nodes.py:584]: Missing comma (elem,)

**Review by tbennun — CHANGES_REQUESTED** [review]

**Review by tbennun — APPROVED** [review]
LGTM, great addition!

**Pull request merged**